File: helloworldservice/src/main/java/helloworldservice/HelloWorldServer.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package helloworldservice;

import org.apache.thrift.TException;
import java.util.Scanner;
/**
 *
 * @author rarora
 */
public class HelloWorldServer implements HelloWorldService.Iface{
    public String hello(String name) throws TException{
        System.out.println(name);
        Scanner sc = new Scanner(System.in);
        while(true){
            String input = sc.nextLine();
            if(input.equals("quit"))
                break;            
        }
        return name + "Hello world from Java server.";
    }
}

File: demo/HelloWorldService/HelloWorldService/HelloWorldServer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HelloWorldService
{
    class HelloWorldServer : HelloWorldService.Iface
    {
        public string hello(string name)
        {
            Console.WriteLine(name + " ");
            return name + ", Hello world from C# server";
        }
    }
}
